### PR TITLE
adds a design document for a broadcast capability

### PR DESCRIPTION
**Pull request by philipwjones**
Describes a design for broadcasting scalars and vectors from one MPI rank to others. Mostly a wrapper around MPI broadcast functions but with a simplified interface for Omega..

### CODE REVIEW

File: components/omega/doc/design/Broadcast.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+so we must be able to broadcast only within these sub-groups using
+the appropriate communicator.
+
+## 3 Algorithmic Formulation
+
+No algorithms needed beyond the MPI broadcast internal algorithm.
+
+## 4 Design
+
+The OMEGA broadcast is mostly just wrappers around the MPI broadcast
```
Comment (sarats) [inline components/omega/doc/design/Broadcast.md:42]: Thinking out loud: We should facilitate nonblocking broadcasts as an option in the framework. 
Example: https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node128.htm#Node128

File: components/omega/doc/design/Broadcast.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+values. 
+  - tests requirement 2.2
+
+### 5.3 Test broadcast from other ranks
+
+Repeat the above two tests except broadcasting from a rank other
+than the master rank.
+  - tests requirement 2.3
+
+### 5.4 Test within a subset environment
```
Comment (sarats) [inline components/omega/doc/design/Broadcast.md:126]: In MPI parlance, does this refer to using an intergroup communicator?

**Comment by sarats** [comment]
If we incorporate a non-blocking broadcast, corresponding `MPI_Wait`s should be invoked at the appropriate stage.

**Review by sarats — APPROVED** [review]

File: components/omega/doc/design/Broadcast.md (hunk 2 above)
Comment (philipwjones) [reply components/omega/doc/design/Broadcast.md:126]: Not in this case - for these functions, they are only broadcasting within the given communicator. So if we created a subset group the call would only broadcast within that group's communicator. There will be a later functionality that will do general communication both within and between groups to rearrange data.

File: components/omega/doc/design/Broadcast.md (hunk 1 above)
Comment (philipwjones) [reply components/omega/doc/design/Broadcast.md:42]: That's a good point. It'll likely be more performant, though I think we've run into some issues in the past when doing too many non-blocking messages when we're broadcasting a lot during init. I'll add an option for it anyway.

**Review by amametjanov — APPROVED** [review]

**Review by brian-oneill — APPROVED** [review]

**Pull request merged**